### PR TITLE
Change CLI to exit on any errors with exit code 1

### PR DESCRIPTION
**Pull request by yux0**
Change CLI tools to throw exit code 1 on errors.
Issue: https://github.com/uber/cadence/issues/706

### CODE REVIEW

File: tools/cli/commands.go [hunk 1]
```diff
@@ -770,7 +765,7 @@ func SignalWorkflow(c *cli.Context) {
 	})
 
 	if err != nil {
-		ErrorAndExit("Signal workflow failed", err)
+		ErrorAndExit("Signal workflow failed.", err)
 	} else {
 		fmt.Println("Signal workflow succeed.")
```
Comment (dmetzgar) [inline tools/cli/commands.go]: Can you put succeeded instead of succeed for these cases?

Comment (yux0) [reply tools/cli/commands.go]: Updated

File: tools/cli/commands.go [hunk 2]
```diff
@@ -205,29 +205,27 @@ var workflowClosedStatusMap = map[string]s.WorkflowExecutionCloseStatus{
...
-}
-
-// ExitIfError exit while err is not nil and print the calling stack also
-func ExitIfError(err error) {
-	const stacksEnv = `CADENCE_CLI_SHOW_STACKS`
 	if err != nil {
-		fmt.Fprintln(os.Stderr, err.Error())
-		if os.Getenv(stacksEnv) != `` {
-			debug.PrintStack()
+		if os.Getenv(showErrorStackEnv) != `` {
```
Comment (vancexu) [inline tools/cli/commands.go]: I am confused, how did you print stack trace?

File: tools/cli/commands.go [hunk 3]
```diff
@@ -369,7 +364,7 @@ func UpdateDomain(c *cli.Context) {
 		if c.IsSet(FlagEmitMetric) {
 			emitMetric, err = strconv.ParseBool(c.String(FlagEmitMetric))
 			if err != nil {
-				ErrorAndExit("Update Domain failed", err)
+				ErrorAndExit(FlagEmitMetric+"format is invalid.", err)
```
Comment (vancexu) [inline tools/cli/commands.go]: missing space

File: tools/cli/commands.go [hunk 4]
```diff
@@ -886,7 +881,7 @@ func DescribeWorkflow(c *cli.Context) {
 // DescribeWorkflowWithID show information about the specified workflow execution
 func DescribeWorkflowWithID(c *cli.Context) {
 	if !c.Args().Present() {
-		ExitIfError(errors.New("workflow_id is required"))
+		ErrorAndExit("Parameter workflow_id is required.", nil)
```
Comment (vancexu) [inline tools/cli/commands.go]: Better to consist with Argument instead of Parameter

File: tools/cli/commands.go [hunk 5]
```diff
@@ -1155,7 +1149,7 @@ func ObserveHistory(c *cli.Context) {
 // ObserveHistoryWithID show the process of running workflow
 func ObserveHistoryWithID(c *cli.Context) {
 	if !c.Args().Present() {
-		ExitIfError(errors.New("workflow_id is required"))
+		ErrorAndExit("workflow_id is required.", nil)
```
Comment (vancexu) [inline tools/cli/commands.go]: Would be better to start all Error msg with Upper case, or add "Argument workflow_id ..."

File: tools/cli/commands.go (hunk 4 above)
Comment (yux0) [reply tools/cli/commands.go]: Updated. Will send out a new PR about the change.

File: tools/cli/commands.go (hunk 5 above)
Comment (yux0) [reply tools/cli/commands.go]: Updated. Will send out a new PR about the change.

File: tools/cli/commands.go (hunk 3 above)
Comment (yux0) [reply tools/cli/commands.go]: Updated. Will send out a new PR about the change.

File: tools/cli/commands.go (hunk 2 above)
Comment (yux0) [reply tools/cli/commands.go]: Faultily thought the error should has stack error. Adding printStack in the new PR.

**Pull request merged**